packages/@aws-cdk/aws-redshift-alpha/lib/cluster.ts

```diff
@@ -584,6 +613,10 @@ export class Cluster extends ClusterBase {
       }
     }
 
+    if (props.resourceAction === ResourceAction.FAILOVER_PRIMARY_COMPUTE && !props.multiAz) {
+      throw new Error('ResourceAction.FAILOVER_PRIMARY_COMPUTE can only be used with multi-AZ clusters.');
+    }
```

Comment:
I tried to execute failover for the single AZ cluster but I got a deployment error.

```ts
    const singleAzCluster = new redshiftAlpha.Cluster(this, 'SingleAzCluster', {
      vpc,
      vpcSubnets: {
        subnetType: ec2.SubnetType.PUBLIC,
      },
      masterUser: {
        masterUsername: 'admin',
      },
      nodeType: redshiftAlpha.NodeType.RA3_XLPLUS,
      publiclyAccessible: true,
    });

    const cfnSingleAzCluster = singleAzCluster.node.defaultChild as redshift.CfnCluster;
    cfnSingleAzCluster.addPropertyOverride('ResourceAction', 'failover-primary-compute');
```

Results:
```sh
8:50:18 | UPDATE_FAILED        | AWS::Redshift::Cluster                      | SingleAzCluster885C9BDD
Resource handler returned message: "Cannot invoke "java.lang.Boolean.booleanValue()" because the return value of "software.amazon.redshift
.cluster.ResourceModel.getMultiAZ()" is null" (RequestToken: d865c8ea-217c-ff00-0c71-1b659fdf6233, HandlerErrorCode: InternalFailure)
```

I also confirmed we can execute failover operation for a multi-AZ cluster.